### ISSUE TITLE
Post-start hooks not always fired

### ISSUE THREAD

**Issue by scornelissen85** [michaelsauter/crane#238]
Sometimes post-start hooks are not fired. I've tested to run a container with Docker 1.9.0 and when I run several times crane run the hook is sometimes fired and sometimes not. For example this yaml:
```
containers:

  test:
    image: centos:7
    run:
      detach: true
      cmd: /bin/sleep 999

hooks:
  test:
    post-start: echo 'hello world'
```
With the command ```crane run test``` executed several times after each other, the hello world is not always fired. Is this some timing issue?

I'm using Docker 1.9.0 with Crane 2.2.0.

**Comment by michaelsauter** [comment]
Hmm interesting. I had an issue with post-start hooks recently but did not have time to look into it yet. So - could very well be an issue with Docker 1.9, or some regression introduced in Crane 2.2.

I'm going to look at it eventually, but no promises about timing ...

Have you tried debugging with `--verbose` to see if all looks good there?

**Comment by scornelissen85** [comment]
Yes, I did use crane -v, and see this result:
```
Using configuration file `/tmp/crane-test.yml`
Command will be applied to: test

Removing container test ...

--> docker rm --force test
test
Running container test ...

--> docker events --filter event=start --filter container=test

--> docker run --detach --name test centos:7 /bin/sleep 999
39f8e8faff17307124db7aeb830928428920bef41c55d27bfbaa694ebcb4fe31

--> echo hello world
hello world
```
And sometimes this:
```
Using configuration file `/tmp/crane-test.yml`
Command will be applied to: test

Removing container test ...

--> docker rm --force test
test
Running container test ...

--> docker events --filter event=start --filter container=test

--> docker run --detach --name test centos:7 /bin/sleep 999
6c71522c575b6105b47bb906ac033a7e8ee50a667a1af06861b791ef55477a0a
```

**Comment by michaelsauter** [comment]
Thanks for investigating and posting this, helps me a lot where to start looking ...

**Comment by bjaglin** [comment]
The `--filter container=test` part of `docker events` seems to be broken in 1.9.0 & 1.9.1, investigating...

**Comment by bjaglin** [comment]
Could not really bisect, trying my luck at https://github.com/docker/docker/issues/18453

**Comment by bjaglin** [comment]
FYI bug confirmed, PR ongoing https://github.com/docker/docker/pull/18464

**Comment by bjaglin** [comment]
Fixed in docker on master via https://github.com/docker/docker/pull/18888 which also changes the output of `docker events` - I guess that should not impact the usage Crane makes of it, right @michaelsauter ?

**Comment by michaelsauter** [comment]
Thanks for the follow-up. I think we should be fine! Hope 1.9.2 is released soon.

**Comment by michaelsauter** [comment]
Looks like 1.9.2 will never be released, but I can confirm 1.10-rc1 fixes the bug and works with Crane.
